test/Elastic.Apm.Tests/SpanTests.cs

```diff
@@ -0,0 +1,32 @@
+using Elastic.Apm.Model;
+using Elastic.Apm.Tests.Mocks;
+using Xunit;
+
+namespace Elastic.Apm.Tests
+{
+	public class SpanTests
+	{
+		[Fact]
+		public void End_ShouldRestoreParentSpan_WhenTransactionIsNotSampled()
+		{
+			// Arrange
+			var currentExecutionSegmentsContainer = new NoopCurrentExecutionSegmentsContainer();
+			var payloadSender = new NoopPayloadSender();
+			var logger = new NoopLogger();
+			var configurationReader = new MockConfigSnapshot();
+
+			var transaction = new Transaction(logger, "transaction", "type", new Sampler(1.0), null, payloadSender, configurationReader,

```

Comment:
As I mentioned at https://github.com/elastic/apm-agent-dotnet/pull/533#pullrequestreview-306361626, we should minimize use of agent's implementation internals in tests and use agent's public API instead.